controllers/suite_test.go

```diff
@@ -108,3 +113,20 @@ func startManager(scheme *runtime.Scheme) {
 		Expect(mgr.Start(stopMgr)).To(Succeed())
 	}()
 }
+
+type fakePodExecutor struct {
+	executedCommands []command
+}
+
+type command []string
+
+func (f *fakePodExecutor) Exec(clientset *kubernetes.Clientset, clusterConfig *rest.Config, namespace, podName, containerName string, command ...string) (string, string, error) {
+	f.executedCommands = append(f.executedCommands, command)
+	return "", "", nil
+}
+
+func (f *fakePodExecutor) ExecutedCommands() []command { return f.executedCommands }
+
+func (f *fakePodExecutor) ResetExecutedCommands() { f.executedCommands = []command{} }
+
+var _ = AfterEach(func() { fakeKubectlExecutor.ResetExecutedCommands() })
```

Comment:
I like that you wrote your own simple to understand fake instead of using a more complex library such as https://github.com/maxbrunsfeld/counterfeiter

Reply:
+1. I love the power that counterfeiter gives you but am not a fan of the amount of boilerplate required, along with the extra step of autogenerating the counterfeit code, just to do something simple like this. Interfaces are powerful for simple fakes :)